### PR TITLE
🐛Fix transparent edges on blurred image placeholders on mobile

### PR DESCRIPTION
**Pull request by addieachan**
- Fixes issue #18113 for amp-img (separate PR for amp-video)

For this image:
screen shot 2018-09-20 at 4 38 59 pm

Old blur:
screen shot 2018-09-20 at 4 38 43 pm

New blur:
screen shot 2018-09-20 at 4 38 16 pm

### CODE REVIEW

**Comment by addieachan** [comment]
@choumx PTAL

File: builtins/amp-img.js [hunk 1]
```diff
@@ -134,6 +140,17 @@ export class AmpImg extends BaseElement {
     guaranteeSrcForSrcsetUnsupportedBrowsers(this.img_);
     this.applyFillContent(this.img_, true);
     this.element.appendChild(this.img_);
+
+    this.fillContent_ = this.element.querySelector('.i-amphtml-fill-content');
```
Comment (choumx) [inline builtins/amp-img.js]: Why not use `this.img_`?

File: builtins/amp-img.js [hunk 2]
```diff
@@ -134,6 +140,17 @@ export class AmpImg extends BaseElement {
...
+    const placeholder = this.getPlaceholder();
+    if (placeholder &&
+      placeholder.classList.contains('i-amphtml-blur') &&
+      isExperimentOn(this.win, 'blurry-placeholder')) {
+      this.blurredPlaceholder_ = placeholder;
+    }
+    if (this.blurredPlaceholder_ && this.fillContent_) {
```
Comment (choumx) [inline builtins/amp-img.js]: When would `this.fillContent_` be null?

File: builtins/amp-img.js [hunk 3]
```diff
@@ -134,6 +140,17 @@ export class AmpImg extends BaseElement {
...
+      setImportantStyles(this.fillContent_, {'visibility': 'hidden'});
```
Comment (choumx) [inline builtins/amp-img.js]: Let's add a comment explaining why we're doing this.

File: builtins/amp-img.js [hunk 4]
```diff
@@ -50,6 +50,12 @@ export class AmpImg extends BaseElement {
 
     /** @private {?UnlistenDef} */
     this.unlistenError_ = null;
+
+    /** @private {?Element} */
+    this.blurredPlaceholder_ = null;
```
Comment (choumx) [inline builtins/amp-img.js]: `blurryPlaceholder_`

**Comment by choumx** [comment]
Can you post a before/after screenshot of this fix?

**Comment by choumx** [comment]
Hmm, might still look weird on non-white backgrounds. Can you grab me to look at this example page together? 

Looks like we're still using the scale transform but it's insufficient here.

**Comment by addieachan** [comment]
Closing for alternate solution (amp-toolbox/pull/96)

**Pull request closed without merging**